src/pugixml.cpp

```diff
@@ -217,6 +217,8 @@ PUGI__NS_BEGIN
 
 	#ifdef PUGIXML_WCHAR_MODE
 		return wcslen(s);
+	#elif defined(PUGIXML_CHAR8_MODE)
+		return strlen(reinterpret_cast<const char*>(s));
```

Comment:
Why is strlength not implemented in terms of `std::char_traits<pugi::char_t>::length`, like you did for the tests?

Reply:
`char_traits` are only available when compiling with STL-support. The tests don't have that restriction.